pkg/daemon/config.go

```diff
@@ -238,6 +255,27 @@ func getDefaultGatewayIface() (string, error) {
 	return "", errors.New("unable to find default route")
 }
 
+func getIfaceOwnPodIP(podIP string) (string, error) {
+	links, err := netlink.LinkList()
+	if err != nil {
+		return "", err
+	}
+
+	for _, link := range links {
+		addrs, err := netlink.AddrList(link, netlink.FAMILY_ALL)
+		if err != nil {
+			return "", fmt.Errorf("failed to get a list of IP addresses %v", err)
+		}
+		for _, addr := range addrs {
+			if addr.IPNet.Contains(net.ParseIP(podIP)) && addr.IP.String() == podIP {
+				return link.Attrs().Name, nil
+			}
+		}
+	}
+
+	return "", errors.New("unable to find podIP interface")
```

Comment:
it can return netlink directly here, no need to use `findInterface` later to find by name